### PR TITLE
Product duplicate Issue

### PR DESCRIPTION
**Pull request by emran-wedevs**

### CODE REVIEW

File: modules/accounting/includes/functions/products.php [hunk 1]
```diff
@@ -102,8 +102,12 @@ function erp_acct_get_product( $product_id ) {
  * Insert product data
  *
  * @param $data
+<<<<<<< HEAD
+ * @return WP_Error
+=======
  *
  * @return int
+>>>>>>> product_duplicate_emran_prev
```
Comment (OpuRockey) [inline modules/accounting/includes/functions/products.php:110]: Please remove unnecessary comments.

File: modules/accounting/includes/functions/products.php [hunk 2]
```diff
@@ -102,8 +102,12 @@ function erp_acct_get_product( $product_id ) {
  * Insert product data
  *
  * @param $data
+<<<<<<< HEAD
```
Comment (OpuRockey) [inline modules/accounting/includes/functions/products.php:105]: Please remove unnecessary comments.

File: modules/accounting/includes/functions/products.php [hunk 3]
```diff
@@ -117,6 +121,19 @@ function erp_acct_insert_product( $data ) {
...
+            $wpdb->prepare(
+                "SELECT * FROM {$wpdb->prefix}erp_acct_products where name = %s",
+                $product_data['name']
+            ),
+            OBJECT
+        );
+
+        if($product_check){
+            throw new \Exception("Product Duplicate problem") ;
+        }
```
Comment (OpuRockey) [inline modules/accounting/includes/functions/products.php:134]: Kindly follow standard practice.

**Review by OpuRockey — COMMENTED** [review]
Please fix those improvements.

File: modules/accounting/includes/functions/products.php [hunk 4]
```diff
@@ -117,6 +118,19 @@ function erp_acct_insert_product( $data ) {
...
+        $product_check =  $wpdb->get_row(
+            $wpdb->prepare(
+                "SELECT * FROM {$wpdb->prefix}erp_acct_products where name = %s",
+                $product_data['name']
+            ),
+            OBJECT
+        );
+
+       if( $product_check ){
+            throw new \Exception("Product Duplicate problem") ;
```
Comment (nurul-umbhiya) [inline modules/accounting/includes/functions/products.php:130]: change to a suitable error message, if you need content related help don't hesitate to contact content team. Also string must be translatable.

File: modules/accounting/includes/functions/products.php [hunk 5]
```diff
@@ -139,11 +153,10 @@ function erp_acct_insert_product( $data ) {
         $wpdb->query( 'COMMIT' );
     } catch ( Exception $e ) {
         $wpdb->query( 'ROLLBACK' );
-
-        return new WP_error( 'product-exception', $e->getMessage() );
+        return new WP_Error( 400,  $e->getMessage() );
     }
 
-    return erp_acct_get_product( $product_id );
+    return  $product_id  ;
```
Comment (nurul-umbhiya) [inline modules/accounting/includes/functions/products.php:159]: Do not change the existing codebase unless necessary, this function might be used by other extensions.

File: modules/accounting/includes/functions/products.php [hunk 6]
```diff
@@ -139,11 +153,10 @@ function erp_acct_insert_product( $data ) {
         $wpdb->query( 'COMMIT' );
     } catch ( Exception $e ) {
         $wpdb->query( 'ROLLBACK' );
-
-        return new WP_error( 'product-exception', $e->getMessage() );
+        return new WP_Error( 400,  $e->getMessage() );
```
Comment (nurul-umbhiya) [inline modules/accounting/includes/functions/products.php:156]: the correct format for sending status code with WP_ERROR is: `return new WP_Error( 'no_posts', __('No post found'), array( 'status' => 404 ) );// status can be changed to any number`

File: modules/accounting/includes/functions/products.php [hunk 7]
```diff
@@ -117,6 +118,19 @@ function erp_acct_insert_product( $data ) {
...
 
+        $product_check =  $wpdb->get_row(
+            $wpdb->prepare(
+                "SELECT * FROM {$wpdb->prefix}erp_acct_products where name = %s",
+                $product_data['name']
+            ),
+            OBJECT
+        );
+
+       if( $product_check ){
```
Comment (nurul-umbhiya) [inline modules/accounting/includes/functions/products.php:129]: fix formatting issue here

File: modules/accounting/includes/functions/products.php [hunk 8]
```diff
@@ -117,6 +118,19 @@ function erp_acct_insert_product( $data ) {
...
+        $product_check =  $wpdb->get_row(
+            $wpdb->prepare(
+                "SELECT * FROM {$wpdb->prefix}erp_acct_products where name = %s",
+                $product_data['name']
+            ),
+            OBJECT
+        );
+
+       if ( $product_check ) {
+            throw new \Exception( __( $product_data['name']." product already exists!" ) ) ;
```
Comment (nurul-umbhiya) [inline modules/accounting/includes/functions/products.php:130]: You missed text domain 👍 
anyways change this line to:
`throw new \Exception( $product_data['name'] . ' ' .  __( "product already exists!" , 'erp) ) ;`

**Review by emran-wedevs — COMMENTED** [review]
Vai, Done everything for this request. You can merge.

File: modules/accounting/includes/functions/products.php (hunk 8 above)
Comment (emran-wedevs) [reply modules/accounting/includes/functions/products.php]: Done

**Pull request merged**